tools/wptrunner/wptrunner/formatters/chromium.py

```diff
@@ -0,0 +1,97 @@
+import json
+import time
+
+from mozlog.formatters import base
+
+
+class ChromiumFormatter(base.BaseFormatter):
+    """Formatter to produce results matching the Chromium JSON Test Results format.
+    https://chromium.googlesource.com/chromium/src/+/master/docs/testing/json_test_results_format.md
+    """
+
+    def __init__(self):
+        # Whether the run was interrupted, either by the test runner or user.
+        self.interrupted = False
+
+        # A map of test status to the number of tests that had that status.
+        self.num_failures_by_status = {}
+
+        # Start time, expressed as offset since UNIX epoch in seconds.
+        self.start_timestamp_seconds = None
+
+        # Trie of test results. Each directory in the test name is a node in
+        # the trie and the leaf contains the dict of per-test data.
+        self.tests = {}
+
+    def _store_test_result(self, name, actual, expected):
+        """
+        Stores the result of a single test in |self.tests|
+        :param str name: name of the test.
+        :param str actual: actual status of the test.
+        :param str expected: expected status of the test.
+        """
+        name_parts = filter(None, name.split("/"))
+        cur_dict = self.tests
+        for name_part in name_parts:
+            if name_part not in cur_dict:
+                cur_dict[name_part] = {}
+            cur_dict = cur_dict[name_part]
+        cur_dict["actual"] = actual
+        cur_dict["expected"] = expected
+
+    def _map_status_name(self, status):
```

Comment:
Is there any reason why this is a function and not just a dict? I don't have an overly strong preference here.

If you keep it as a function can you add a failure case that throws if the status is unknown?

Reply:
Added a default result, which I chose to just return the status as-is. Not sure which would be better quite yet, this may change after this logger actually gets used for a bit.